src/scripts/main.js

```diff
@@ -1 +1,76 @@
 'use strict';
+
+function createMessage(text, className = 'success') {
+  const div = document.createElement('div');
+
+  div.classList = className;
+  div.setAttribute('data-qa', 'notification');
+  div.textContent = text;
+  document.body.append(div);
+}
+
+function addClickHandlers(mouseClick) {
+  document.addEventListener('click', mouseClick);
+  document.addEventListener('contextmenu', mouseClick);
+}
+
+function removeClickHandlers(mouseClick) {
+  document.removeEventListener('click', mouseClick);
+  document.removeEventListener('contextmenu', mouseClick);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+      removeClickHandlers(handleClick);
+    }
+  };
+
+  addClickHandlers(handleClick);
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+    removeClickHandlers(handleClick);
+  }, 3000);
```

Comment:
Consider adding `clearTimeout` to cancel the timeout if the promise is resolved before the timeout occurs. This will prevent unnecessary rejection if the promise is already resolved.